src/components/KModal/KModal.vue

```diff
@@ -100,343 +87,270 @@
   actionButtonText: {
     type: String,
     default: 'Submit',
   },
-  /**
-  * Set the appearance of the action/proceed button
-  */
   actionButtonAppearance: {
     type: String as PropType<ButtonAppearance>,
     default: 'primary',
   },
-  /**
-  * Set the text of the close/cancel button
-  */
+  actionButtonDisabled: {
+    type: Boolean,
+    default: false,
+  },
+  hideCancelButton: {
+    type: Boolean,
+    default: false,
+  },
   cancelButtonText: {
     type: String,
     default: 'Cancel',
   },
-  /**
-  * Set the appearance of the close/cancel button
-  */
   cancelButtonAppearance: {
     type: String as PropType<ButtonAppearance>,
-    default: 'secondary',
+    default: 'tertiary',
   },
-  /**
-  * Set to not render the cancel button
-  */
-  hideCancelButton: {
+  cancelButtonDisabled: {
     type: Boolean,
     default: false,
   },
-  /**
-  * Options to be passed to tabbable
-  */
   tabbableOptions: {
     type: Object,
     default: () => ({}),
   },
-  /**
-  * Test mode - for testing only, strips out generated ids
-  */
-  testMode: {
-    type: Boolean,
-    default: false,
-  },
-  /**
-   * The max-width of the modal. **Default: `500px`**.
-   */
-  maxWidth: {
+  width: {
     type: String,
     required: false,
     default: '500px',
   },
+  maxHeight: {
+    type: String,
+    required: false,
+    default: 'calc(100vh - 200px)',
+  },
+  closeOnBackdropClick: {
+    type: Boolean,
+    default: false,
+  },
+  proceedOnEnter: {
+    type: Boolean,
+    default: true,
+  },
+  hideCloseIcon: {
+    type: Boolean,
+    default: false,
+  },
 })
 
 const emit = defineEmits<{
-  (e: 'canceled'): void
+  (e: 'cancel'): void
   (e: 'proceed'): void
 }>()
 
+const attrs = useAttrs()
 const slots = useSlots()
-const focusTrap = ref<InstanceType<typeof FocusTrap> | null>(null)
-const modalOuter = ref<{ $el: HTMLElement} | null>(null)
 
-const hasHeaderImage = computed((): boolean => {
-  return !!slots['header-image']
-})
+const focusTrapElement = ref<InstanceType<typeof FocusTrap> | null>(null)
+const modalWrapperElement = ref<HTMLElement | null>(null)
 
-const dismissButtonColor = computed((): string => {
-  if (props.dismissButtonTheme === 'light') {
-    return `var(--kui-color-text-neutral-weak, ${KUI_COLOR_TEXT_NEUTRAL_WEAK})`
-  }
+const widthValue = computed((): string => getSizeFromString(props.width))
+const maxHeightValue = computed((): string => getSizeFromString(props.maxHeight))
+
+const sanitizedAttrs = computed(() => {
+  const attributes = Object.assign({}, attrs)
 
-  return `var(--kui-color-text-neutral-stronger, ${KUI_COLOR_TEXT_NEUTRAL_STRONGER})`
+  delete attributes.class // delete  because we bind it to the class
+
+  return attributes
+})
+
+const showHeader = computed((): boolean => {
+  return !!props.title || !!slots.title || !props.hideCloseIcon
 })
 
-const handleKeydown = (e: any): void => {
-  if (props.isVisible && e.keyCode === 27) {
+const handleKeydown = (event: any): void => {
```

Comment:
This event is too "greedy" and listens for `Enter` no matter what element in the DOM is focused.

The `Enter` keypress should only be triggered if the modal or one of its children are focused.

Reply:
Changed to only have event listener when modal is open